server/app/models/story_code.py

```diff
@@ -30,6 +34,11 @@ def use(self):
         if self.single_use:
             self.used = True
             self.put()
+        #effectively last used for multi-use codes
+        self.used_date = datetime.datetime.now()
```

Comment:
Let's move the self.put() from line 36 to after we set the used_date; otherwise we're probably not saving the new used_date >_>